pkg/controller/scylladbmonitoring/sync_grafana.go

```diff
@@ -0,0 +1,502 @@
+	return grafanav1alpha1assets.GrafanaServiceTemplate.RenderObject(map[string]any{
+		"scyllaDBMonitoringName": sm.Name,
+	})
+}
+
+func makeGrafanaIngress(sm *scyllav1alpha1.ScyllaDBMonitoring) (*networkingv1.Ingress, string, error) {
+	ingressOptions := getGrafanaIngressOptions(sm)
+	if ingressOptions == nil {
+		return nil, "", nil
+	}
+
+	if ingressOptions.Disabled != nil && *ingressOptions.Disabled == true {
+		return nil, "", nil
+	}
+
+	if len(ingressOptions.DNSDomains) == 0 {
+		return nil, "", nil
+	}
+
+	return grafanav1alpha1assets.GrafanaIngressTemplate.RenderObject(map[string]any{
+		"scyllaDBMonitoringName": sm.Name,
+		"dnsDomains":             ingressOptions.DNSDomains,
+		"ingressAnnotations":     ingressOptions.Annotations,
+		"ingressClassName":       ingressOptions.IngressClassName,
+	})
+}
+
+func (smc *Controller) syncGrafana(
+	ctx context.Context,
+	sm *scyllav1alpha1.ScyllaDBMonitoring,
+	configMaps map[string]*corev1.ConfigMap,
+	secrets map[string]*corev1.Secret,
+	services map[string]*corev1.Service,
+	serviceAccounts map[string]*corev1.ServiceAccount,
+	deployments map[string]*appsv1.Deployment,
+	ingresses map[string]*networkingv1.Ingress,
+) ([]metav1.Condition, error) {
+	var progressingConditions []metav1.Condition
+
+	grafanaServingCertChainConfig := &okubecrypto.CertChainConfig{
+		CAConfig: &okubecrypto.CAConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   fmt.Sprintf("%s-grafana-serving-ca", sm.Name),
+				Labels: getGrafanaLabels(sm),
+			},
+			Validity: 10 * 365 * 24 * time.Hour,
+			Refresh:  8 * 365 * 24 * time.Hour,
+		},
+		CABundleConfig: &okubecrypto.CABundleConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   fmt.Sprintf("%s-grafana-serving-ca", sm.Name),
+				Labels: getGrafanaLabels(sm),
+			},
+		},
+		CertConfigs: []*okubecrypto.CertificateConfig{
+			{
+				MetaConfig: okubecrypto.MetaConfig{
+					Name:   fmt.Sprintf("%s-grafana-serving-certs", sm.Name),
+					Labels: getGrafanaLabels(sm),
+				},
+				Validity: 30 * 24 * time.Hour,
+				Refresh:  20 * 24 * time.Hour,
+				CertCreator: (&ocrypto.ServingCertCreatorConfig{
+					Subject: pkix.Name{
+						CommonName: "",
+					},
+					IPAddresses: nil,
+					DNSNames: append(
+						[]string{
+							sm.Name + "-grafana",
+						},
+						getGrafanaIngressDomains(sm)...,
+					),
+				}).ToCreator(),
+			},
+		},
+	}
+
+	var certChainConfigs okubecrypto.CertChainConfigs
+
+	spec := getGrafanaSpec(sm)
+
+	var grafanaServingCertSecretName string
+	if spec != nil {
+		grafanaServingCertSecretName = spec.ServingCertSecretName
+	}
+
+	if len(grafanaServingCertSecretName) == 0 {
+		grafanaServingCertSecretName = grafanaServingCertChainConfig.CertConfigs[0].Name
+		certChainConfigs = append(certChainConfigs, grafanaServingCertChainConfig)
+	}
+
+	// Render manifests.
+	var renderErrors []error
+
+	requiredGrafanaSA, _, err := grafanav1alpha1assets.GrafanaSATemplate.RenderObject(map[string]any{
```

Comment:
nit: extract it to a function to match every other resource

Reply:
done